embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/IndexTaskTest.java

```diff
@@ -99,6 +99,9 @@ public void test_runIndexTask_forInlineDatasource()
     }
 
     cluster.callApi().waitForAllSegmentsToBeAvailable(dataSource, coordinator, broker);
+    broker.latchableEmitter().waitForMetricEvent(
+        event -> event.hasDimension(DruidMetrics.DATASOURCE, dataSource)
+    );
```

Comment:
Is this needed? the call before this line (`waitForAllSegmentsToBeAvailable`) should already be taking care of segment availability.